tests/assembly/cmse.rs

```diff
@@ -1,6 +1,9 @@
+//@ revisions: hard soft
 //@ assembly-output: emit-asm
-//@ compile-flags: --target thumbv8m.main-none-eabi --crate-type lib -Copt-level=1
-//@ needs-llvm-components: arm
+//@ [hard] compile-flags: --target thumbv8m.main-none-eabihf --crate-type lib -Copt-level=1
+//@ [soft] compile-flags: --target thumbv8m.main-none-eabi --crate-type lib -Copt-level=1
+//@ [hard] needs-llvm-components: arm
+//@ [soft] needs-llvm-components: arm
```

Comment:
Suggestion: you can factor this out and just have

```
//@ needs-llvm-components: arm
```

which will unconditionally apply to all revisions.

Reply:
We tried this initially but it did complain about it

Reply:
Ah right, tidy requires this be paired with `--target` I think. Nevermind.